### PR TITLE
Feat/dirty form modal

### PR DESCRIPTION
**Pull request by tnodell**
https://app.clickup.com/t/2704425/ID-5977

Handles a user clicking a link which would navigate away from a dirty detail panel.

https://github.com/exogee-technology/graphweaver/assets/81122022/da2a332f-8e78-4464-9358-27e109127049

### CODE REVIEW

File: src/examples/s3-storage/src/frontend/types.ts [hunk 1]
```diff
@@ -17,8 +17,8 @@ export type Scalars = {
   Boolean: { input: boolean; output: boolean; }
   Int: { input: number; output: number; }
   Float: { input: number; output: number; }
-  /** Media type scalar */
-  Media: { input: any; output: any; }
+  /** Image type scalar */
```
Comment (tnodell) [inline src/examples/s3-storage/src/frontend/types.ts:20]: These type changes are corrections from the demo. Apologies they got caught up in this PR. This is correct though.

File: src/packages/admin-ui-components/src/detail-panel/component.tsx [hunk 2]
```diff
@@ -348,32 +362,71 @@ export const DetailPanel = () => {
...
+				isOpen={!!modalRedirectUrl}
+				hideCloseX
+				modalContent={
+					<>
+						<div className={styles.unsavedChangesTitle}>You have unsaved changes</div>
+						<div className={styles.unsavedChangesContent}>
+							Are you sure you want to leave this entity? Your changes will not be saved.
+						</div>
+						<div className={styles.unsavedChangesButtonRow}>
+							<Button className={styles.unsavedChangesButton} onClick={handleCancelLeave}>
```
Comment (siowhoi) [inline src/packages/admin-ui-components/src/detail-panel/component.tsx:419]: Can you get latest from main? I had updated the style for reset type. Can you use reset type button? `<Button type="reset" onClick={handleCancelLeave}>
								Cancel
							</Button>
							<Button type="reset" onClick={handleConfirmLeave}>
								Discard Changes
							</Button>`

**Review by siowhoi — APPROVED** [review]
Looks good.

File: src/packages/admin-ui-components/src/detail-panel/component.tsx (hunk 2 above)
Comment (tnodell) [reply src/packages/admin-ui-components/src/detail-panel/component.tsx:419]: Oh ya will do, thanks!

Comment (tnodell) [reply src/packages/admin-ui-components/src/detail-panel/component.tsx:419]: ah no these don't match my modal designs... thank you though.

**Pull request merged**